canopen/node/local.py

```diff
@@ -115,6 +117,15 @@ def set_data(
         self.data_store.setdefault(index, {})
         self.data_store[index][subindex] = bytes(data)
 
+        if 0x1800 <= index <= 0x19FF:
+            # TPDO Communication parameter changed
+            tpdoNum = (index - 0x1800) + 1
+            self._tpdo_configuration_write(tpdoNum)
+        elif 0x1A00 <= index <= 0x1BFF:
+            # TPDO Mapping parameter changed
+            tpdoNum = (index - 0x1A00) + 1
+            self._tpdo_configuration_write(tpdoNum)
```

Comment:
This could be shortened a bit because the ranges are adjacent.  You only need one range check and one modulo to find the TPDO number.  But feel free to ignore this if you think it's not an improvement.